python/cuspatial/cuspatial/bindings/trajectory.pyx

```diff
@@ -1,68 +1,86 @@
     cdef gdf_column* c_trajectory_id = <gdf_column*>malloc(sizeof(gdf_column))
-    cdef gdf_column* c_len = <gdf_column*>malloc(sizeof(gdf_column))
+    cdef gdf_column* c_length = <gdf_column*>malloc(sizeof(gdf_column))
     cdef gdf_column* c_pos = <gdf_column*>malloc(sizeof(gdf_column))
 
     with nogil:
          num_trajectories = derive_trajectories(c_x[0], c_y[0],
                                                 c_object_id[0],
                                                 c_timestamp[0],
                                                 c_trajectory_id[0],
-                                                c_len[0], c_pos[0])
+                                                c_length[0], c_pos[0])
 
     traj_id_data, traj_id_mask = gdf_column_to_column_mem(c_trajectory_id)
-    len_data, len_mask = gdf_column_to_column_mem(c_len)    
+    length_data, length_mask = gdf_column_to_column_mem(c_length)
     pos_data, pos_mask = gdf_column_to_column_mem(c_pos)
     trajectory_id = Column.from_mem_views(traj_id_data,
                                           traj_id_mask)
-    len = Column.from_mem_views(len_data, len_mask)
+    length = Column.from_mem_views(length_data, length_mask)
     pos = Column.from_mem_views(pos_data, pos_mask)
 
-    return num_trajectories, trajectory_id, len, pos
+    return (num_trajectories,
+            DataFrame({'trajectory_id': Series(trajectory_id),
+                       'length': Series(length),
+                       'position': Series(pos)}))
 
-cpdef cpp_trajectory_distance_and_speed(x, y, timestamp, len, pos):
+
+cpdef cpp_trajectory_distance_and_speed(x, y, timestamp, length, pos):
+    x = x.astype('float64')._column
+    y = y.astype('float64')._column
+    timestamp = timestamp.astype('datetime64[ms]')._column
+    length = length.astype('int32')._column
+    pos = pos.astype('int32')._column
     cdef gdf_column* c_x = column_view_from_column(x)
     cdef gdf_column* c_y = column_view_from_column(y)
     cdef gdf_column* c_timestamp = column_view_from_column(timestamp)
-    cdef gdf_column* c_len = column_view_from_column(len)
+    cdef gdf_column* c_length = column_view_from_column(length)
     cdef gdf_column* c_pos = column_view_from_column(pos)
     cdef pair[gdf_column, gdf_column] c_distance_speed
 
     with nogil:
         c_distance_speed = trajectory_distance_and_speed(c_x[0], c_y[0],
                                                          c_timestamp[0],
-                                                         c_len[0],c_pos[0])
+                                                         c_length[0],c_pos[0])
 
     dist_data, dist_mask = gdf_column_to_column_mem(&c_distance_speed.first)
     speed_data, speed_mask = gdf_column_to_column_mem(&c_distance_speed.second)
     dist=Column.from_mem_views(dist_data, dist_mask)
     speed=Column.from_mem_views(speed_data, speed_mask)
 
-    return dist,speed
+    return Series(dist), Series(speed)
```

Comment:
Why not make this a DataFrame?

Reply:
No disagreement! I'm working in that direction as I get better bringing the two APIs together.